api/v1alpha1/pkisecretengineconfig_types.go

```diff
@@ -0,0 +1,535 @@
+	// +kubebuilder:validation:Optional
+	PostalCode string `json:"postalCode,omitempty"`
+
+	// Specifies the Serial Number, if any. Otherwise Vault will generate a random serial for you. If you want more than one, specify alternative names in the alt_names map using OID 2.5.4.5.
+	// +kubebuilder:validation:Optional
+	SerialNumber string `json:"serialNumber,omitempty"`
+}
+
+type PKIConfig struct {
+	// +kubebuilder:validation:Optional
+	PKIConfigUrls `json:",inline"`
+	// +kubebuilder:validation:Optional
+	PKIConfigCRL `json:",inline"`
+	// // +kubebuilder:validation:Optional
+	// PKIConfigCA `json:",inline"`
+}
+
+type PKIConfigUrls struct {
+	// Specifies the URL values for the Issuing Certificate field. This can be an array or a comma-separated string list.
+	// +kubebuilder:validation:Optional
+	// +listType=set
+	// kubebuilder:validation:UniqueItems=true
+	IssuingCertificates []string `json:"issuingCertificates,omitempty"`
+
+	// Specifies the URL values for the CRL Distribution Points field. This can be an array or a comma-separated string list.
+	// +kubebuilder:validation:Optional
+	// +listType=set
+	// kubebuilder:validation:UniqueItems=true
+	CRLDistributionPoints []string `json:"CRLDistributionPoints,omitempty"`
+
+	// Specifies the URL values for the OCSP Servers field. This can be an array or a comma-separated string list.
+	// +kubebuilder:validation:Optional
+	// +listType=set
+	// kubebuilder:validation:UniqueItems=true
+	OcspServers []string `json:"ocspServers,omitempty"`
+}
+
+type PKIConfigCA struct {
+	// Specifies the key and certificate concatenated in PEM format.
+	// +kubebuilder:validation:Required
+
+	PemBundle string `json:"pemBundle,omitempty"`
+}
+
+type PKIConfigCRL struct {
+	// Specifies the time until expiration.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="72h"
+	CRLExpiry metav1.Duration `json:"CRLExpiry,omitempty"`
+
+	// Disables or enables CRL building.
+	// +kubebuilder:validation:Optional
+	CRLDisable bool `json:"CRLDisable,omitempty"`
+}
+
+type PKIIntermediate struct {
+	// Secret retrieves the signed intermediate certificate from a Kubernetes secret. Allows submitting the signed CA certificate corresponding to a private key generated.
+	// +kubebuilder:validation:Optional
+	Secret *corev1.LocalObjectReference `json:"secret,omitempty"`
+
+	// CertificateKey key to be used when retrieving the signed certificate
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="tls.crt"
+	CertificateKey string `json:"certificateKey,omitempty"`
+
+	// Use the configured refered Vault PKISecretEngineConfig to issue a certificate with appropriate values for acting as an intermediate CA.
+	// +kubebuilder:validation:Optional
+	InternalSign *corev1.LocalObjectReference `json:"internalSign,omitempty"`
+
+	CSR string `json:"-"`
+
+	SignedIntermediate string `json:"-"`
```

Comment:
if this is a private field, it should not be exported.

Reply:
is it not the correct syntax? Or just remove the JSON part?